datafusion/sql/src/expr/function.rs

```diff
@@ -90,6 +90,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             let partition_by = window
                 .partition_by
                 .into_iter()
+                //ignore window spec PARTITION BY for scalar values
+                .filter(|e| !matches!(e, sqlparser::ast::Expr::Value { .. },))
                 .map(|e| self.sql_expr_to_logical_expr(e, schema, planner_context))
                 .collect::<Result<Vec<_>>>()?;
             let mut order_by = self.order_by_to_sort_expr(
```

Comment:
According to test it seems that we have no problem with ScalarValues inside window `order_by`. However, I think we can apply similar filtering done for `window.partition_by` to the `window.order_by`.